client/src/features/modal/EventModal.jsx

```diff
@@ -9,6 +9,36 @@ import dataService from "../../services/dataService";
 import { useAuthContext } from "../../contexts/AuthContext";
 import { useEventsContext } from "../../contexts/EventsContext";
 
+const URLRegex = /(https?:\/\/[^\s]+|www\.[^\s]+)/g;
+
+const LinkText = ({ children }) => {
+  if (typeof children !== "string")
+    throw new Error("LinkText can only accept a string as children");
+
+  const separated = children.split(URLRegex);
+
+  return separated.map((res, index) => {
+    if (URLRegex.test(res)) {
+      // Prepend http:// if it starts with www. to make a valid href
+      const href = res.startsWith("www.")
+        ? `http://${res}`
+        : new URL(res).toString();
+      return (
+        <a
+          key={index}
+          href={href}
+          target="_blank"
+          rel="noopener noreferrer"
+          className="text-blue-600 underline hover:text-blue-800"
+        >
+          {res}
```

Comment:
This should reference the href as display links can be deliberately misleading.

```suggestion
          {href}
```